src/attributes/classes.js

```diff
@@ -24,19 +22,17 @@ jQuery.fn.extend({
 			});
 		}
 
-		if ( proceed ) {
-			// The disjunction here is for better compressibility (see removeClass)
-			classes = ( value || "" ).match( rnotwhite ) || [];
+		if ( typeof value === "string" && value ) {
+			classes = value.match( rnotwhite ) || [];
 
-			for ( ; i < len; i++ ) {
-				elem = this[ i ];
+			while ( ( elem = this[ i++ ] ) ) {
```

Comment:
I like the `while` better too, but there are places where changing to it from `for` will introduce failures on collections including falsy members ([`.find`](https://github.com/Mr21/jquery/blob/classes-better-compressibility/src/traversing/findFilter.js#L55-L76), [`.closest`](https://github.com/Mr21/jquery/blob/classes-better-compressibility/src/traversing.js#L72), etc.). I'm suggesting that we do so anyway (as is being done here) and document that such collections are invalid.